### ISSUE TITLE
[12] Order Book Improvements

### ISSUE THREAD

**Issue by wmbutler** [bitshares/bitshares-ui#1062]
The order book is a little busy. It's not really wide enough to warrant striped rows. Let's remove those. Also, if we can give a reference to the depth chart here with significant markers, it may remove the need for the depth chart toggle. This allows the user to have more information without toggling. This is a rough start and intended to spark discussion. I also noticed that most charting systems only show the size of the order for one side of the market.

I.e., if we are on BTS/USD, everything is priced in USD.

<img width="423" alt="screen shot 2018-01-23 at 7 47 23 pm" src="https://user-images.githubusercontent.com/1254810/35310434-fc4ddb56-0076-11e8-9306-de32570d8242.png">

**Comment by vanissoft** [comment]
Nice look, although big depth chart is a need.
+1 to be priced in base asset.

**Comment by fractalnode** [comment]
we can not turn this large portion of information into such a small portion

![info](https://user-images.githubusercontent.com/20418259/35471173-2be6cd3c-0356-11e8-9365-e6a594c5f96f.png)

**Comment by kapeer42** [comment]
Maybe this answers your concerns (work in progress)
![peek 2018-01-27 15-05](https://user-images.githubusercontent.com/35779885/35472676-e15ac368-0373-11e8-88bd-02cdd86a6638.gif)

**Comment by wmbutler** [comment]
We can still keep the toggle for the Market Depth. I won't use it, but no reason to get rid of the option.

**Comment by abitmore** [comment]
I don't think it's a good idea to remove the `total value` column from the order book.
1. It's very convenient for traders,
2. due to the implementation in core, prices showing on the book are usually inaccurate.

**Comment by tbone-bts** [comment]
I agree with @abit re: cumulative quantity column on the order book. It will be very inconvenient if you remove it.

**Comment by wmbutler** [comment]
When you say "Total Value", what are you referring to? All I see is the asset quantity in each side of the trade.

![screen shot 2018-01-28 at 1 31 11 pm](https://user-images.githubusercontent.com/1254810/35486155-a15c4144-042f-11e8-84b3-6c63bae6ff1a.png)

**Comment by tbone-bts** [comment]
Your screenshot shows three columns:  (1) price (which for clarity should really be labeled bid/ask),  (2) quantity of the asset being traded (in this case the quantity of BTS), and (3) price x quantity (i.e. total value of the bid or ask in terms of the base asset, in this case USD).  

Also, you're looking at the vertical order book, which is ok for just watching a market...but not great for placing trades.  The horizontal order book, on the other hand, has a 4th column displaying the cumulative running total of the value all bids (or asks) up to and including that row.  That is very, **VERY** useful. 

>When you say "Total Value", what are you referring to? All I see is the asset quantity in each side of the trade. As far as cumulative quantity, that doesn't exist in the order book as you can see from this screenshot.

screen shot 2018-01-28 at 1 34 29 pm

**Comment by wmbutler** [comment]
This issue isn't proposing to make changes to the horizontal order book. Let's not confuse things. My points remain the same:

- There is no cumulative value in the existing vertical order book
- It's quite standard across exchanges to only display pricing in the purchase currency

**Comment by abitmore** [comment]
> When you say "Total Value", what are you referring to? All I see is the asset quantity in each side of the trade. As far as cumulative quantity, that doesn't exist in the order book as you can see from this screenshot.

I mean the screenshot in OP shows only 2 columns. I like 3.

**Comment by wmbutler** [comment]
I'm not sure it makes sense to show 3. Looking across many different trading platforms, this not accepted practice and I think it adds to confusion when new users hit our platform.

**Comment by abitmore** [comment]
Top crypto exchanges:
1. binance, showing 3: https://www.binance.com/trade.html?symbol=ETH_BTC
2. OKEx, horizontal layout, showing 2: https://www.okex.com/spot.html
More others to be added here.

**Comment by wmbutler** [comment]
Point taken. Thanks.

- [ ] Keep pricing in both assets.

**Comment by abitmore** [comment]
@calvinfroedge what's the progress so far? If you haven't started working on it, can you release it so I can ask @ety001 to work on it after finished #1106?

**Comment by wmbutler** [comment]
This needs to be worked on with @ahdigital as well. He is doing a full UX on the Exchange. #1449

**Comment by abitmore** [comment]
@wmbutler I feel that @ahdigital is overloaded, in the meanwhile we have quite some developers that can code now. Please prioritize things.

**Comment by abitmore** [comment]
Seems it has been implemented here: https://www.hellobts.com/market/BTS_CNY

**Comment by sschiessl-bcp** [comment]
@startailcoon 

Has this been picked up for your restyling? If not, I could evaluate after you are done and work out if anything should be added from here.

**Comment by wmbutler** [comment]
I submitted a proposed style as the first comment here. I like it because it allows the order book to serve two roles. @startailcoon would love to see where you end up with this.

**Comment by MikhailKavalenka** [comment]
@startailcoon , Do we only need to show the depth chart representation on the order book sections in vertical mode? Or the scope of the issue is more then that?

**Comment by startailcoon** [comment]
@MikhailKavalenka for me it would make sense to have this in the orderbook, regardless of position of it. So I would say it should be both cases.
